app/workers/open_all_courses_on_apply_worker.rb

```diff
@@ -0,0 +1,17 @@
+class OpenAllCoursesOnApplyWorker
+  include Sidekiq::Worker
+
+  def perform
+    return unless CycleTimetable.after_apply_reopens?
```

Comment:
@duncanjbrown added this to try and prevent the worker from being run if apply hasn't reopened i.e. not in the 2022 cycle. Not sure if as a result this has made it quite restrictive, thoughts?

Reply:
It might be worth either

1. removing it 
2. CycleTimetable.after_apply_reopens? - 1.day so we can do it the evening before the deadline

Reply:
Thinking more about this. We will have to do this after Find reopens as the `ApplicationForm.current_cycle' will be scoped to 2021 until after find opens in the timetable.

Reply:
or

```ruby
return unless RecruitmentCycle.current_year == 2022
```

as this won't be used again next year

Reply:
have chucked that in ☝️